### ISSUE TITLE
ArgumentException when parsing a version

### ISSUE THREAD

**Issue by kiwipiet** [stevedesmond-ca/dotnet-libyear#17]
Getting the following exception when trying to parse the version of System.IdentityModel.Tokens.Jwt
Using Libyear: 3.0.0

`<package id="System.IdentityModel.Tokens.Jwt" version="4.0.4.403061554" targetFramework="net461" />`

```
Sorry, an unexpected exception has occurred: System.ArgumentException: '4.0.4.403061554' is not a valid version string.
Parameter name: value
   at NuGet.Versioning.SemanticVersion.Parse(String value)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at LibYear.Lib.FileTypes.XmlProjectFile..ctor(String filename, String elementName, String[] packageAttributeNames, String versionAttributeName) in D:\Code\LibYear\src\LibYear.Lib\FileTypes\XmlProjectFile.cs:line 29
   at LibYear.Lib.FileTypes.PackagesConfigFile..ctor(String filename) in D:\Code\LibYear\src\LibYear.Lib\FileTypes\PackagesConfigFile.cs:line 7
   at LibYear.Lib.ProjectFileManager.<>c.<FindProjectsInDir>b__3_4(FileInfo f) in D:\Code\LibYear\src\LibYear.Lib\ProjectFileManager.cs:line 36
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at System.Linq.Set`1.UnionWith(IEnumerable`1 other)
   at System.Linq.Enumerable.UnionIterator`1.FillSet()
   at System.Linq.Enumerable.UnionIterator`1.ToList()
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at LibYear.App.Runner.Run(IReadOnlyList`1 args) in D:\Code\LibYear\src\LibYear\Runner.cs:line 34
   at LibYear.App.Program.Main(String[] args) in D:\Code\LibYear\src\LibYear\Program.cs:line 24

```

**Comment by kiwipiet** [comment]
I tried resolving this by changing the dependency to a newer version, now I'm getting:
`Sorry, an unexpected exception has occurred: System.ArgumentException: '1.0.0.0' is not a valid version string.`

**Comment by stevedesmond-ca** [comment]
Thanks for this bug report -- I believe the official NuGet library being used to parse versions only supports 3 part semantic versioning.

It's on my list to upgrade to the latest version of the NuGet libraries, which may be able to handle this case, but I have the feeling it's more likely going to be a case of no longer parsing semantic versions, which I'm not really inclined to do.

If that's the case, I think maybe just skipping unsupported version formats is probably preferable.

**Comment by stevedesmond-ca** [comment]
Resolved by #20, will get a new release out shortly -- thanks, @mchughmk!
